clippy fixes

jfinkels opened the pull request:
Found using the following command (which appears in `.pre-commit.yaml`):
```
cargo +nightly clippy --workspace --all-targets --all-features --
```

sylvestre commented on src/uu/echo/src/echo.rs:
```diff
@@ -59,6 +59,10 @@ fn parse_code(
+// Clippy complatins that `.peek()` is never called on
+// `iter`. However, it is called inside the `parse_code()` function
+// that borrows `iter`.
+#[allow(clippy::unused_peekable)]
```
we are getting this here:
```
ERROR: `cargo clippy`: unknown lint: `clippy::unused_peekable` (file:'src/uu/echo/src/echo.rs', line:65)
```

I guess we don't use the latest version in the CI

jfinkels replied:
I'll just remove this change for now.

sylvestre commented on src/uu/echo/src/echo.rs:
```diff
@@ -59,6 +59,10 @@ fn parse_code(
     std::char::from_u32(ret)
 }
 
+// Clippy complatins that `.peek()` is never called on
```
complains ? :)

tertsdiepraam commented:
Maybe we should remove the `+nightly` from the `.pre-commit.yaml`? I think it's best to wait for the lints to stabalize and have the false positives removed.

jfinkels commented:
> Maybe we should remove the `+nightly` from the `.pre-commit.yaml`? I think it's best to wait for the lints to stabalize and have the false positives removed.

That's fine with me, I don't have any strong opinion about this.

I have updated this pull request to remove the unstable `#[allow()]` and corrected the spelling.
